### DISCUSSION TITLE
A proper way of using embedding as feature matrix

### DISCUSSION THREAD

**Discussion by JihoChoi** [pyg-team/pytorch_geometric#5315] [Q&A] [answered]
What is the proper example of using the output (embedding) of Transformer (or LSTM) as the input feature matrix?

Each node in the graph consists of a `sequence of elements` and the representation of the sequence trained in RNNs.
I would like to build an end-to-end model in which the output of the RNNs is the input the to GNNs as node representations.
However, I believe, all the examples consist of static feature matrix X.
What would be the proper example of passing the embedding to the input?

```
from torch_geometric.data import Data

data = Data(
    x                   = x_embedding,    # output tensor from the transfomer
    y                   = y_label,
    edge_index = edge_index,
    train_mask = mask,
)
```

**Comment by rusty1s** [comment, marked as answer]
It depends on whether you want to fine-tune your transformer/LSTM or simply want to use their embeddings (detached from the computation graph). In case of the latter, you can simply input your embeddings into `data.x` (as you have done). In case of the former, you would need to input the sequence into the transformer as part of your model. Does that answer your question? :)

**Reply by JihoChoi** [reply]
Thank you very much!
